sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -2371,31 +2371,41 @@ def __GetBodiesFromQueryResult(result):
             feedrangeEPK = partition_key_definition._get_epk_range_for_prefix_partition_key(partition_key)  # cspell:disable-line # pylint: disable=line-too-long
             over_lapping_ranges = await self._routing_map_provider.get_overlapping_ranges(id_, [feedrangeEPK])
             results = None
+            # For each over lapping range we will take a sub range of the feed range EPK that overlaps with the over
+            # lapping physical partition. The EPK sub range will be one of three:
+            # 1) Will have a range min equal to the feed range EPK min, and a range max equal to the over lapping
+            # partition
+            # 2) Will have a range min equal to the over lapping partition range min, and a range max equal to the
+            # feed range EPK range max.
+            # 3) will match exactly with the current over lapping physical partition, so we just return the over lapping
+            # physical partition's partition key id.
             for over_lapping_range in over_lapping_ranges:
-                # It is possible for the over lapping range to include multiple physical partitions
-                # we should return query results for all the partitions that are overlapped.
                 single_range = routing_range.Range.PartitionKeyRangeToRange(over_lapping_range)
-                if single_range.min == feedrangeEPK.min and single_range.max == feedrangeEPK.max:
-                    # The EpkRange spans exactly one physical partition
+                # Since the range min and max are all Upper Cased string Hex Values,
+                # we can compare the values lexicographically
+                EPK_sub_range = routing_range.Range(range_min=max(single_range.min, feedrangeEPK.min),
```

Comment:
nit: Consider moving this logic somewhere common if it's going to be reused? To avoid having to remember to touch 2 places if there is a fix to be done